core/controllers/learner_dashboard.py

```diff
@@ -101,3 +102,35 @@ def get(self):
             'subscription_list': subscription_list
         })
         self.render_json(self.values)
+
+
+class LearnerDashboardFeedbackThreadHandler(base.BaseHandler):
+    """Gets all the messages in a thread."""
+
+    def get(self, exploration_id, thread_id):
+        """Handles GET requests."""
+        if self.user_id is None:
+            raise self.PageNotFoundException
+
+        messages = feedback_services.get_messages(
+            exploration_id, thread_id)
+        author_ids = [m.author_id for m in messages]
+        authors_settings = user_services.get_users_settings(author_ids)
+
+        message_ids = [m.message_id for m in messages]
+        feedback_services.update_messages_read_by_the_user(
+            exploration_id, thread_id, self.user_id, message_ids)
```

Comment:
Style nit: Odd that self.user_id is nestled between the other args. Maybe put it at the front? It seems like exp_id, thread_id and message_ids should all come together.